### PR TITLE
bt-257: create onboarding step2

### LINKED ISSUE #257: Create Onboarding step2 - BSA badges
- [ ] Add to bundle component bsa-badges-step 
- [ ] On Next button data should be validated?, use useAppForm() hook for this
- [ ] After validation save data in the Store

Image

**User story:**
As a talent, I want to be able to choose and add badges to my profile, so that potential employers can get familiar with my soft skills

**Acceptance criteria:**
“Your average project score”, “Your average lectures score”, “Level of English” should be chosen by default and talent can’t change it
Talent should be able to choose only “Communication score”, “Working with team score”, “Punctuality” badges
Talent should be able to choose several badges
Talent should be able to go to the next step without choosing badges
Talent should be able to back to the previous step "Profile" by clicking on the “Back” button
Talent should be able to go to the next step "Skills and projects" by clicking on the “Next” button
Talent 's information should be saved in the fields when switching between steps

### PR DESCRIPTION
**Pull request by savinaDaria**
#257 
image

### CODE REVIEW

File: frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import { FormHelperText } from '@mui/material';
```
Comment (nika-sirak) [inline frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx]: change import { FormHelperText} from '~/bundles/common/components/components.js';

File: frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import {
+    Controller,
+    type ControllerFieldState,
+    type ControllerRenderProps,
+    type UseFormStateReturn,
+} from 'react-hook-form';
+import { useSelector } from 'react-redux';
```
Comment (nika-sirak) [inline frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx]: We have hook for useSelector
import { useAppSelector} from '~/bundles/common/hooks/hooks.js';

File: frontend/src/bundles/lms/mock-data/mock-badges.constants.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/lms/mock-data/mock-badges.constants.ts:1]: we will use this data only in the onboarding. I think we can keep it in the `talent-onboarding` bundle

File: frontend/src/bundles/talent-onboarding/context/form-submit-provider.context.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    >;
+};
+
+type ProviderProperties = {
+    children?: React.ReactNode;
+};
+
+const FormSubmitContext = createContext<FormSubmitContextType | null>(null);
+
+// Custom hook to use the context
```
Comment (nikita-remeslov) [inline frontend/src/bundles/talent-onboarding/context/form-submit-provider.context.tsx]: why do we need this hook?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/talent-onboarding/context/form-submit-provider.context.tsx (hunk 3 above)
Comment (savinaDaria) [reply frontend/src/bundles/talent-onboarding/context/form-submit-provider.context.tsx]: @nikita-remeslov I used it here to test step data saving onSubmit, but actually it is added in _combine talent onboarding PR_
I'll comment/delete stuff related to it

File: frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+//TODO: uncomment when context for submitting will be added
+const BsaBadgesStep: React.FC = () => {
+    const savedPayload = useSelector(
+        (state: RootReducer) => state.talentOnBoarding.bsaBadgesStep,
+    );
+
+    const bsaBadges = useSelector(
+        //TODO: get all user badges here, for now mock data
+        (state: RootReducer) => state.lms.bsaBadges,
+    );
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/talent-onboarding/components/bsa-badges-step/bsa-badges-step.tsx]: I think you don't need two useSelector's here.

Maybe this approach will be better

```
const { savedPayload, bsaBadges } =
    useAppSelector((state) => ({
      savedPayload: state.talentOnBoarding.bsaBadgesStep,
      bsaBadges: state.lms.bsaBadges,
    }));
```

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Review by nika-sirak — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**